src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -339,18 +339,25 @@ public void ProcessEvaluationFinishedEventArgs(
 
             _buildEventsProcessor
                 .ProcessEvaluationFinishedEventArgs(checkContext, evaluationFinishedEventArgs, propertiesLookup);
+
+            CheckScopeClassifier.RaiseNotifyOnScopingReadiness();
```

Comment:
For the in-proc node and worker nodes this happens after first evaluation is done:

https://github.com/dotnet/msbuild/blob/f422d8d7dfe0a7115b11b31470215ad6b7723138/src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs#L1168

How about adding the state+event to `FileClassifier` - there flipping the state on first `RegisterKnownImmutableLocations` call, and in the `CheckScopeClassifier`  just passing through the chack and event subscription to the `FileClassifier`?

Something like (pseudocode):
```csharp
internal class FileClassifier
{
   private bool _isInitialized;
   public bool IsInitialized {get => _isInitialized; private set { if(!_isInitialized){ /* invoke event */ } _isInitialized = value; } }
   public event Action? OnInitialized;

   public void RegisterKnownImmutableLocations(/*...*/)
  {
    // ...
   this.IsInitialized = true;
  }
}

internal static class CheckScopeClassifier
{
   internal bool IsInitialized => FileClassifier.Shared.IsInitialized;
   internal static event Action? OnInitialized
  {
    add => FileClassifier.Shared.OnInitialized += value;
    remove => FileClassifier.Shared.OnInitialized -= value;
  }
}
```